wiki/Article_Style_Guide/en.md

```diff
@@ -1,192 +1,318 @@
+
+#### Serial comma
+
+When listing items of 3 or more in a sentence, use the serial comma.
+
+- The game modes of osu!: osu!standard, osu!taiko, osu!catch**,** and osu!mania are fun to play with others.
+
+## Code
+
+Use code (inline) when noting:
 
-2.  In-game terms should be treated as common nouns and as such *should not* be capitalised. Proper nouns, however, such as the different game mode names (Standard, Taiko, Catch the Beat and Mania) or the mascots' names (Pippi, Yuzu, Maria) *should* always be capitalised.
+- a key on the keyboard
+  - i.e. `B`, `Alt` or `Ctrl` + `Shift` + `A`
+- a name of a button
+  - i.e. `1. Let's do it.` or `4. Noooo! I didn't mean to!`
+- a name of a folder
+  - i.e. `Exports` or `Downloads`
+- a name of a directory
+  - i.e. `/Songs/` or `/Skins/User/`
+- a name of a keyword
+  - i.e. `ComboBurstStyle` or `[Events]`
+- the name of a file extension
+  - i.e. `.jpg` or `.osz`
 
-#### Game modes  
+When noting a keyboard key, use capital letters for single characters and [Camel case](https://en.wikipedia.org/wiki/Camel_case) for modifers.
 
-The name of game modes should always be capitalised unless if it is following the exclamation mark without a blank space of the name *osu!*.
+When naming a file extension, add a period (`.`) then the file extension in lower cased letters.
 
-Example:
+## Links
 
+Article links must follow these rules:
 
-*osu!mania is a game mode that has been introduced years later the original game mode that is osu! Standard.*
+1. must use relative-absolute paths, unless linking to subfolder (then use relative-relative paths)
+  - i.e. `[Welcome](/wiki/Welcome/)`
+  - i.e. (from `Game_Modes` page) `[osu!mania](./osu!mania)`
+2. must be spelt correctly (links are case sensitive!!)
+3. must not specify the specific file name **regardless of language** (osu!wiki will handle this)
+  - do: `[Glossary](/wiki/Glossary/)`
+  - don't: `[Glossary](/wiki/Glossary/fr.md)`
+4. must not link to section headings (section headings on osu!wiki does not work)
 
-<!-- -->
+External links must follow these rules:
 
-#### Game Modifiers  
+- prefer the `https://` protocol
+- if referencing another site, must link to a reputable source
+- must be a clean and direct link
+  - do not use a third-party shortened link
+  - do not use links that link to ads
+  - do not use a url with long url queries, parameters, or fragments
+- the link name must be the title text of the page it is linking to
 
-Game modifiers as well as their abbreviations *should* always be capitalised.
+Do use use the word `here` as the link text.
+As it may seem convenient to use, it could be misleading.
+Instead, take the page's title (from the top of page, or from the `<title>` tag) and use that as the link text.
 
-Examples:
+You can use either the reference or inline style links.
+If you are using the reference style linking, it is sugguested to place the reference links at the top of the article for quicker access.
```

Comment:
For link maintenance.
Pretty sure I will be _pissed_ if I have to scroll down to the end of the page just to fix the links.

YMMV so no comment.